pkg/apis/common/v1/common.go

```diff
@@ -70,21 +82,39 @@ func (o ObjectSelector) WithDefaultNamespace(defaultNamespace string) ObjectSele
 		Namespace:   defaultNamespace,
 		Name:        o.Name,
 		ServiceName: o.ServiceName,
+		SecretName:  o.SecretName,
 	}
 }
 
+func (o ObjectSelector) NameOrSecretName() string {
+	name := o.Name
+	if o.SecretName != "" {
+		name = o.SecretName
+	}
+	return name
+}
+
 // NamespacedName is a convenience method to turn an ObjectSelector into a NamespacedName.
 func (o ObjectSelector) NamespacedName() types.NamespacedName {
 	return types.NamespacedName{
-		Name:      o.Name,
+		Name:      o.NameOrSecretName(),
 		Namespace: o.Namespace,
 	}
 }
 
-// IsDefined checks if the object selector is not nil and has a name.
+// IsDefined checks if the object selector is not nil and has a name or a secret name.
 // Namespace is not mandatory as it may be inherited by the parent object.
 func (o *ObjectSelector) IsDefined() bool {
-	return o != nil && o.Name != ""
+	return o != nil && o.NameOrSecretName() != ""
+}
+
+// IsObjectTypeSecret returns true when the object selector references a Kubernetes secret describing the referenced object.
+func (o ObjectSelector) IsObjectTypeSecret() bool {
```

Comment:
That makes sense. What about `IsUnmanaged()` as I went in this direction with `GetUnmanagedAssociationConnectionInfoFromSecret()` and `ReconcileUnmanagedAssociation`? Or it could be confusing with https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-troubleshooting-methods.html#k8s-exclude-resource?

I've a little reservation to abstract the secret notion because callers uses this function to make something with the secret. Mut maybe it's ok to say it's an implementation detail to know about associations with external references. Otherwise I was thinking of something like `IsSecretForExternal` or `IsSecretForUnmanaged`.